### PR TITLE
written function to create reference

### PR DESCRIPTION
**Pull request by bhanuc**
I have written the javascript implementation of create reference and updated the readme. My coffeescript is rusty so I have skipped that part. Would like to know your views before I contribute more.

### CODE REVIEW

**Comment by pksunkara** [comment]
> My coffeescript is rusty so I have skipped that part

That isn't how this repo works. You need to write the CoffeeScript and then generate the JS.

**Comment by bhanuc** [comment]
I just wanted to get a review of my code . btw added the coffee source and
compiled js with coffee as asked by you

File: README.md [hunk 1]
```diff
@@ -1220,6 +1220,14 @@ ghsearch.code({
 }, callback); //array of search results
 ```
 
+#### Create Reference
+
+```js
+ghrepo.create_reference('ref','sha', function(err,result){
+  console.log(err,result)
+}); //If no error then reference has been created
```
Comment (pksunkara) [inline README.md]: No need for the whole thing. Just change this to

```js
ghrepo.createReference('master', '18293abcd72', callback);
```

File: src/octonode/repo.coffee [hunk 2]
```diff
@@ -140,6 +140,15 @@ class Repo
       return cb(err) if err
       if s isnt 200 then cb(new Error("Repo branch error")) else cb null, b, h
 
+  # Get a reference for a repository
+  create_reference: (newbranch, sha, cb) ->
```
Comment (pksunkara) [inline src/octonode/repo.coffee]: Change the function name to `createReference`

File: src/octonode/repo.coffee [hunk 3]
```diff
@@ -140,6 +140,15 @@ class Repo
...
+    if typeof newbranch is 'string' and typeof sha is 'string' and typeof cb is 'function'
```
Comment (pksunkara) [inline src/octonode/repo.coffee]: We don't need this `if` loop here.

File: src/octonode/repo.coffee [hunk 4]
```diff
@@ -140,6 +140,15 @@ class Repo
...
+      @client.post "/repos/#{@name}/git/refs", { ref : "refs/heads/#{newbranch}", "sha": sha}, (err, s, b, h) ->
+        return cb(err) if err
+        if s isnt 201 then cb(new Error("Repo branch Create error.")) else cb null, b, h
```
Comment (pksunkara) [inline src/octonode/repo.coffee]: `new Error("Repo createReference error")`

File: src/octonode/repo.coffee [hunk 5]
```diff
@@ -140,6 +140,15 @@ class Repo
       return cb(err) if err
       if s isnt 200 then cb(new Error("Repo branch error")) else cb null, b, h
 
+  # Get a reference for a repository
+  create_reference: (newbranch, sha, cb) ->
+    if typeof newbranch is 'string' and typeof sha is 'string' and typeof cb is 'function'
+      @client.post "/repos/#{@name}/git/refs", { ref : "refs/heads/#{newbranch}", "sha": sha}, (err, s, b, h) ->
```
Comment (pksunkara) [inline src/octonode/repo.coffee]: `{ref: "refs/heads/#{ref}", sha: sha}`

File: src/octonode/repo.coffee (hunk 2 above)
Comment (pksunkara) [inline src/octonode/repo.coffee]: Please change `newbranch` to `ref`

Comment (pksunkara) [inline src/octonode/repo.coffee]: Add a comment before this line as

```
# '/repos/pksunkara/hub/git/refs' POST
```

File: README.md [hunk 6]
```diff
@@ -1220,6 +1220,14 @@ ghsearch.code({
 }, callback); //array of search results
 ```
 
+#### Create Reference
```
Comment (pksunkara) [inline README.md]: This should be moved to be after https://github.com/pksunkara/octonode#get-a-branch-for-a-repository-get-repospksunkarahubbranchesmaster.

Also, please change this line to the following:

```
#### Create a Reference (POST /repos/pksunkara/hub/git/refs)
```

**Comment by pksunkara** [comment]
Reviewed. Please do the suggested changes, use CoffeeScript 1.9.3 to generate JS and then squash the commits.

The code and documentation should follow the style patterns from other functions which this PR does not.

Thanks :smile:

**Comment by bhanuc** [comment]
Do I need to CoffeeScript 1.9.3 or 1.10.0 would suffice ?

**Comment by pksunkara** [comment]
As said above, please use 1.9.3

**Comment by bhanuc** [comment]
please check now.

File: src/octonode/repo.coffee [hunk 7]
```diff
@@ -140,6 +140,13 @@ class Repo
       return cb(err) if err
       if s isnt 200 then cb(new Error("Repo branch error")) else cb null, b, h
 
+  # Get a reference for a repository
+  # '/repos/pksunkara/hub/git/refs' POST
+  createReference: (ref, sha, cb) ->
+    @client.post "/repos/#{@name}/git/refs", {ref: "refs/heads/#{ref}", sha: sha}, (err, s, b, h) ->
+        return cb(err) if err
+        if s isnt 201 then cb(new Error("Repo createReference error")) else cb null, b, h
```
Comment (pksunkara) [inline src/octonode/repo.coffee]: Please use 2 space indentation for the above 2 lines.

**Comment by pksunkara** [comment]
Just one minor issue left. And please squash the commits into a single commit after fixing it.

**Comment by bhanuc** [comment]
Done !! Thanks for the guiding me through. Next time will surely be a smoother ride.

**Pull request merged**